libmscore/edit.cpp

```diff
@@ -1053,9 +1053,26 @@ void Score::regroupNotesAndRests(const Fraction& startTick, const Fraction& endT
                                     continue; // this voice is empty here
                               if (!cr->isRest() || s->tick() + cr->actualTicks() > maxTick || toRest(cr)->isGap())
                                     break; // next element in the same voice is not a rest, or it exceeds the selection, or it is a gap
+                              if (cr->tuplet()) {
+                                    Tuplet* tuplet = cr->tuplet(); // element is a tuplet; special case
+                                    short tupletSize = tuplet->elements().size();
+                                    bool tupletAllRests = true;
+                                    for (auto const& el : tuplet->elements()) {
+                                          // Check all tuplet elements
+                                          if (!el->isRest())
+                                                tupletAllRests = false; // Tuplet element is not a rest
+                                        }
+                                    if (!tupletAllRests) break;
+                                    // Set to last element (i.e. rest) of the tuplet
+                                    cr = toRest(tuplet->elements().back());
+                                    for (int i = 0; i < tupletSize - 1; i++) {
+                                          // Skip the rest of the segments belonging to the tuplet
+                                          s = s->next(SegmentType::ChordRest);
```

Comment:
I don't think you can rely on the fact that a tuplet of N elements will cover exactly N chordrest segments.  There may, after all, be additional notes on other staves.

Reply:
t is better better indeed to get the first segment after the tuplet and this can be found by adding the duration of the tuplet to the position of the first element in the segment and find the segment on that new position.

Reply:
This has been rewritten. Not exactly the solution @njvdberg suggested, but I think this should do the job. Otherwise, please explain what 'position' would correspond to.